catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/RemoveAllCommand.java

```diff
@@ -236,35 +220,27 @@ private boolean needsAlternateQueryAndResponse(SourceResponse response) {
         return false;
     }
 
-    boolean isAccidentalRemoval(PrintStream console) throws IOException {
+    private boolean isAccidentalRemoval() {
         if (!force) {
-            StringBuffer buffer = new StringBuffer();
-
-            //use a message specific to whether they
-            //are removing from cache or the catalog
-            String warning = (this.cache ?
-                    WARNING_MESSAGE_FORMAT_CACHE_REMOVAL :
-                    WARNING_MESSAGE_FORMAT_CATALOG_REMOVAL);
-            System.err.println(String.format(warning, (expired ? "expired " : "")));
-            System.err.flush();
+            //use a message specific to cache and expired options
+            final String warning = String.format(
+                    "WARNING: This will permanently remove all %1$srecords from the %2$s. Do you want to proceed? (yes/no): ",
+                    (expired ? "expired " : ""),
+                    (cache ? "cache" : "Catalog"));
             while (true) {
-                int byteOfData = session.getKeyboard()
-                        .read();
-
-                if (byteOfData < 0) {
-                    // end of stream
+                final String response;
+                try {
+                    response = session.readLine(warning, null);
+                } catch (IOException e) {
+                    console.println("Please use \"removeall -f\" or \"removeall --force\" instead");
                     return true;
                 }
-                System.err.print((char) byteOfData);
-                if (byteOfData == '\r' || byteOfData == '\n') {
-                    break;
+                if (response.equalsIgnoreCase("yes")) {
+                    return false;
+                } else if (response.equalsIgnoreCase("no")) {
```

Comment:
I think we might want to make this else/if just an else so if we don't match 'yes' we treat it as a no. Or maybe print something to the console that tells the user the input was invalid.